syntax = "proto3";

import "google/protobuf/timestamp.proto";

service Chirper {
  rpc Command(PRequest) returns (PReply) {}
  rpc BracketOrder(OrderRequest) returns (OrderReply) {}
  rpc AllOrdersStatus(PRequest) returns (Orders) {}
  rpc LiveData(SRequest) returns (OHLCVs) {}
  rpc ChildOrdersStatus(PRequest) returns (ChildOrders) {}
}
message SRequest {
  string symbol = 1;
  string exchange = 2;
}
message OHLCVs {
  repeated OHLCV ohlcv = 1;
}
message OHLCV {
  google.protobuf.Timestamp time = 1;
  int32 pOpen = 2;
  int32 pHigh = 3;
  int32 pLow = 4;
  int32 pClose = 5;
  int32 volume = 6;
}
enum Status {
  OPEN = 0;
  COMPLETE = 1;
}
message Orders {
  repeated Order order = 1;
}
message Order {
  google.protobuf.Timestamp ordertime = 1;
  string symbol = 2;
  Status status = 3;
  repeated ChildOrder childorder = 4;
}
message ChildOrders {
  repeated ChildOrder childorder = 1;
}

message ChildOrder {
  string orderno = 1;
  string status = 2;
  int32 p5Price = 3;
}

enum Type {
  BUY = 0;
  SELL = 1;
}

message OrderRequest {
  string symbol = 1;
  Type type = 2;
  int32 p5Price = 3;
  int32 p5StopLoss = 4;
  int32 p5Target = 5;
}
message OrderReply {
  google.protobuf.Timestamp ordertime = 1;
}
message PRequest {
  string msg = 1;
}

message PReply {
  string msg = 1;
}
